### ISSUE TITLE
Implement GPT training loop

### ISSUE THREAD

**Issue by Cyber-fungus07** [Cyber-fungus07/mini-GPT#1]
Implement the training pipeline for the GPT model using cross-entropy loss.
Tasks:

1.  [x] Create training loop
2.  [x] Create validation loop
3.  [x] Add optimizer
4.  [x] Track training and validation loss
5.  [x] Add configurable number of epochs
6.  [x] Add checkpoint saving
7.  [x] Test training on a small dataset
